### ISSUE TITLE
processArguments not used with noReset:true

### ISSUE THREAD

**Issue by owlsnakes** [appium/appium#13020]
## The problem

The AUT process launchArguments to configure in app behavior.
They are used with noReset:false but not noReset:true

## Environment

* appium 1.14.0
* Desktop OS/version used to run Appium: macOS Mojave version 10.14.6
* Node.js version (unless using Appium.app|exe):v12.1.0
* Npm or Yarn package manager: npm 6.9.0
* Mobile platform/version under test: iOS 11.4
* Real device or emulator/simulator: iPhone SE
* Appium CLI or Appium.app|exe: Appium CLI

## Details

Our app uses a launch arguments to configure in-app settings.
The launch arguments are not used when appium launches the app with noReset:false

**Comment by mykola-mokhnach** [comment]
`launchArguments` only make sense if one starts/restarts the app under test. noReset prevents this behaviour though. What would be your proposal?

**Comment by owlsnakes** [comment]
Hello!
oh ok. so it's not as simple as I hoped.

Is there any way to launch an app with launch arguments *without*
reinstalling it? - that would be the desired outcome if possible.

**Comment by mykola-mokhnach** [comment]
xcuitest driver itself does support setting arguments and/or environments vars for launchApp call. although, this is not supported in clients: https://github.com/appium/appium-xcuitest-driver/blob/fe9f6e238cd676fdaa18a6cff86b786c44349e47/lib/commands/app-management.js#L58

**Comment by owlsnakes** [comment]
is the lack of support because it's impossible or it just hasn't been done
yet?

**Comment by mykola-mokhnach** [comment]
Nothing complicated there. It just has to be done. You are welcome to push a PR for the client that you use

**Comment by owlsnakes** [comment]
Great thank you Mykola! I do appreciate the time you spent talking to me.
Have a good weekend =)

**Comment by mykola-mokhnach** [comment]
Actually, you could do the trick even now by providing the corresponding options to `mobile: launchApp` endpoint. For example:

```python
driver.execute_script('mobile: launchApp', {
    'bundleId': 'com.example',
    'arguments': ['arg1', arg2],
    'environment': {
       'var1': 'value1'
    }
})
```

**Comment by mykola-mokhnach** [comment]
@owlsnakes does the suggestion above work for you?

**Comment by owlsnakes** [comment]
I don't think so;
https://github.com/facebookarchive/WebDriverAgent/issues/919

**Comment by mykola-mokhnach** [comment]
Can you attach the actual server logs? I don't understand how this issue is related to application startup.

**Comment by owlsnakes** [comment]
This seems to be resolved in Appium 1.15.1
